makefiles/dependencies_debug.inc.mk

```diff
@@ -0,0 +1,51 @@
+# Debug targets to evaluate dependency migrations
+#
+# The goal is to get the value of variables used for dependency resolution
+# and the impact of refactoring
+
+# Files output can be generated through `info-boards-supported` evaluation with
+#
+#     DEPENDENCY_DEBUG=1 make info-boards-supported
+#
+# And for each board in the normal make context with
+#
+#     BOARD=board_name make dependency_debug
+#
+# To compare in an aggregated file, you can run in an application directory:
+#
+#    for board in $(make info-boards); do DEPENDENCY_DEBUG_OUTPUT_DIR=bin/info BOARD=${board} make dependency_debug; done; cat bin/info/* > bin/deps_info
+#    DEPENDENCY_DEBUG=1 DEPENDENCY_DEBUG_OUTPUT_DIR=bin/info-global make info-boards-supported; cat bin/info-global/* > bin/deps_info-boards-supported
+#    # And compare both files
+#    diff -u bin/deps_info bin/deps_info-boards-supported
+#
+# And when comparing two revisions, include the revision in the file names
+
+.PHONY: dependency_debug
+# Only generate the dependencies when the board is not disabled
+# This will allow comparing with the output of `info-boards-supported` more easily
+dependency_debug:
```

Comment:
Just a style remark, why `"_"` most recipe we have use `"-"` as far as I can tell.

Reply:
Indeed, will change.